ott/geometry/costs.py

```diff
@@ -121,7 +122,7 @@ def pairwise(self, x: jnp.ndarray, y: jnp.ndarray) -> float:
     y_norm = jnp.linalg.norm(y, axis=-1)
     cosine_similarity = jnp.vdot(x, y) / (x_norm * y_norm + ridge)
     cosine_distance = 1.0 - cosine_similarity
-    return cosine_distance
+    return jnp.clip(cosine_distance, 0., 2.)
```

Comment:
why the clip here? maybe a comment?

Reply:
I found some numerical imprecisions that cause this to be negative, will add a comment.